Sources/BranchSDK/NSError+Branch.m

```diff
@@ -115,4 +115,50 @@ + (BOOL)branchDNSBlockingError:(NSError *)error {
     return NO;
 }
 
++ (BOOL)branchVPNBlockingError:(NSError *)error {
+    if (error) {
+        NSError *underlyingError = error.userInfo[@"NSUnderlyingError"];
+        if (underlyingError) {
+
+            /**
+             `Domain=kCFErrorDomainCFNetwork Code=-1004` indicates that the connection failed because a connection can't be made to the host.
+             Reference: https://developer.apple.com/documentation/cfnetwork/cfnetworkerrors/kcfurlerrorcannotconnecttohost?language=objc
+             
+             `_kCFStreamErrorCodeKey=61` indicates that the connection was refused.
+             Reference: https://opensource.apple.com/source/xnu/xnu-792/bsd/sys/errno.h.auto.html
+             */
+            
+            BOOL isCouldntConnectErrorCode = [@(-1004) isEqual:@(underlyingError.code)];
+            BOOL isLocalHostErrorKey = [@(61) isEqual:error.userInfo[@"_kCFStreamErrorCodeKey"]];
+            
+            if ([self isConnectedToVPN] && isCouldntConnectErrorCode && isLocalHostErrorKey) {
+                return YES;
+            }
+        }
+    }
+    return NO;
+}
+
+/**
+ Helper method to which checks the device's internet proxy settings for common VPN protocol and interface substrings to determine if a VPN enabled.
+ https://developer.apple.com/documentation/cfnetwork/cfnetworkcopysystemproxysettings()
+ */
++ (BOOL)isConnectedToVPN {
+    NSDictionary *proxySettings = (__bridge NSDictionary *)(CFNetworkCopySystemProxySettings());
+    if (proxySettings) {
+        NSDictionary *scopedSettings = proxySettings[@"__SCOPED__"];
+        if (scopedSettings) {
+            for (NSString *key in scopedSettings) {
+                if ([key containsString:@"tap"] ||
+                    [key containsString:@"tun"] ||
+                    [key containsString:@"ppp"] ||
+                    [key containsString:@"ipsec"]) {
```

Comment:
How/why are these keys checked?

Reply:
These are common VPN protocols:
- IPsec: Ipsec uses encryption, encapsulating an IP packet inside an IPsec packet. De-encapsulation happens at the end of the tunnel, where the original IP packet is decrypted and forwarded to its intended destination
- Tun (or Utun)/Tap: Serve similar tunneling purposes, but they operate on different layers of the network stack. More info [Here](https://en.wikipedia.org/wiki/TUN/TAP)
-PPP: Stands for “Point-to-Point Protocol.” PPP is a protocol that enables communication and data transfer between two points or “nodes”. Since it encapsulates other protocols, it can be used for data tunneling (Full definition [Here](https://techterms.com/definition/ppp))